src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallerFactory.cs

```diff
@@ -40,14 +42,17 @@ internal class WorkloadInstallerFactory
                     nugetPackageDownloader, verbosity, packageSourceLocation, reporter, tempDirPath);
             }
 
-            if (elevationRequired && !CanWriteToDotnetRoot(dotnetDir))
+            if (elevationRequired && !WorkloadInstall.IsUserLocal(dotnetDir, sdkFeatureBand.ToString()) && !CanWriteToDotnetRoot(dotnetDir))
```

Comment:
It's still possible that the user local directory will require admin permissions to access, right? So this check is probably still valuable.